internal/http/logs_sanitize_test.go

```diff
@@ -28,142 +31,150 @@ func TestSanitizeLogFile(t *testing.T) {
 		},
 		{
-			input:    "Cerberus identify us3rn4me P1D",
-			expected: "Cerberus identify us3rn4me REDACTED",
+			input:    "\"module\":\"irc\" erica letmeinannounce us3rn4me IRCKEY",
+			expected: "\"module\":\"irc\" erica letmeinannounce us3rn4me REDACTED",
 		},
 		{
-			input:    "NickServ IDENTIFY dasøl13sa#!",
-			expected: "NickServ IDENTIFY REDACTED",
+			input:    "\"module\":\"irc\" Cerberus identify us3rn4me IRCKEY",
+			expected: "\"module\":\"irc\" Cerberus identify us3rn4me REDACTED",
 		},
 		{
-			input:    "--> AUTHENTICATE poasd!232kljøasdj!%",
-			expected: "--> AUTHENTICATE REDACTED",
+			input:    "\"module\":\"irc\" NickServ IDENTIFY Nvbkødn~vzjHkPEimnJ6PmJw8ayiE#wg",
+			expected: "\"module\":\"irc\" NickServ IDENTIFY REDACTED",
+		},
+		{
+			input:    "\"module\":\"irc\" PRIVMSG NickServ IDENTIFY zAPEJEA8ryYnpj3AiE3KJ",
+			expected: "\"module\":\"irc\" PRIVMSG NickServ IDENTIFY REDACTED",
 		},
 	}
 
 	for _, testCase := range testCases {
-		// Create a temporary file with sample log data
-		tmpFile, err := ioutil.TempFile("", "test-log-*.log")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer os.Remove(tmpFile.Name())
+		t.Run(testCase.name, func(t *testing.T) {
+			// Create a temporary file with sample log data
+			tmpFile, err := ioutil.TempFile("", "test-log-*.log")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer os.Remove(tmpFile.Name())
+
+			// Write the test case input to the temporary file
+			_, err = tmpFile.WriteString(testCase.input + "\n")
+			if err != nil {
+				tmpFile.Close()
+				t.Fatal(err)
+			}
+			err = tmpFile.Close()
+			if err != nil {
+				t.Fatal(err)
+			}
 
-		// Write sample log data to the temporary file
-		_, err = tmpFile.WriteString(testCase.input)
-		if err != nil {
-			tmpFile.Close()
-			t.Fatal(err)
-		}
-		err = tmpFile.Close()
-		if err != nil {
-			t.Fatal(err)
-		}
+			// Call SanitizeLogFile on the temporary file
+			sanitizedContent, err := SanitizeLogFile(tmpFile.Name())
+			if err != nil {
+				t.Fatal(err)
+			}
 
-		// Call SanitizeLogFile on the temporary file
-		sanitizedTmpFilePath, err := SanitizeLogFile(tmpFile.Name())
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer os.Remove(sanitizedTmpFilePath)
+			// Read the content of the sanitized content
+			buf := new(bytes.Buffer)
+			_, err = buf.ReadFrom(sanitizedContent)
+			if err != nil {
+				t.Fatal(err)
+			}
 
-		// Read the content of the sanitized temporary file
-		sanitizedData, err := ioutil.ReadFile(sanitizedTmpFilePath)
-		if err != nil {
-			t.Fatal(err)
-		}
+			sanitizedData := buf.String()
 
-		// Check if the sanitized data matches the expected content
-		if string(sanitizedData) != testCase.expected {
-			t.Errorf("Sanitized data does not match expected data for input: %s\nExpected:\n%s\nActual:\n%s", testCase.input, testCase.expected, sanitizedData)
-		}
+			// Check if the sanitized data matches the expected content
+			if !strings.Contains(sanitizedData, testCase.expected+"\n") {
```

Comment:
Check the length, HasPrefix, then check the last character with another conditional to avoid the memory allocation.

Reply:
> Check the length, HasPrefix, then check the last character with another conditional to avoid the memory allocation.

It's a test case, no need.

Reply:
There's a couple of us on Discord if you'd like to chat in real time.